# Mutual Inductance and Transformers
### *Chapter 4 Section 7* -- AC Steady State Analysis

The previous analysis of a coil assumed that the only flux linking a coil was that due to its
own current, and consequently, the only voltage induced was that due to this current. In this
Chapter coupled coils and phenomenon of mutual inductance will be discussed. A two-port
equations expressing voltages by currents, both in time-domain and frequency-domain, will
be considered. Then, the most practical use of this phenomenon in transformers will be
studied. Basic transformer built of practical coils will be considered at first, next an ideal
transformer. Finally the ideal transformer based model of practical transformer will be
described.

## Mutual Inductance – Basic Transformer
Consider two coupled coils, as shown in Fig. 4.7.1.  

![Fig-4.7.1](Figures/Fig-4.7.1.png)

The current $i_1$ produces in coil 1 flux $\phi_{11}$. Part of this flux threads coil 2, the remainder is coil 1's leakage flux. They are denoted '$\phi_{21}$' and '$\phi_{l1}$' respectively.  Similarly, $i_2$ produces in coil 2 flux $\phi_{22}$ that is split into two fluxes, $\phi_{12}$ and $\phi_{l2}$.  

\begin{equation}\tag{1a}
	\phi_{11} = \phi_{21} + \phi_{l1}
\end{equation}

\begin{equation}\tag{1b}
	\phi_{22} = \phi_{12} + \phi_{l2}
\end{equation}

Electric analog of this magnetic circuit is presented in Fig. 4.7.2  

![Fig-4.7.2](Figures/Fig-4.7.2.png)

Via KCL,  

\begin{equation}\tag{2.1}
    \text{a: }\;\phi_{11} + \phi_{12} - \phi_{l1} = \phi \quad\quad\text{and}\quad\quad \text{b: }\;\phi = \phi_{22} + \phi_{21} - \phi_{l2}
\end{equation}

Via the above two equations and equivalance principle,  

$$ \phi_{11} + \phi_{12} - \phi_{l1} = \phi_{22} + \phi_{21} - \phi_{l2}$$

Via the above equation and equations (1a,b),  

$$ \phi_{21} + \phi_{12} = \phi_{12} + \phi_{21}$$

The electric analog in figure 4.7.2 is thus consistent with equations (1a,b).

Via KVL and Ohm's Law,  

\begin{equation}\tag{2.2}
    \text{a: }\;U_{l1} = \phi_{l1}R_{l1} = i_1z_1 \quad\quad \text{b: }\; U_m = R_m\phi = i_1z_1 + i_2z_2 \quad\quad\text{c: }\; i_2z_2 = \phi_{l2}R_{l2} = U_{l2}
\end{equation}

And, via equations (2.1a) and (1b), or equations (2.1b) and (1a),    

\begin{equation}\tag{2.3}
    \phi = \phi_{11} + \phi_{22} - \phi_{l1} - \phi_{l2}
\end{equation}

where $R_{l1}$ and $R_{l2}$ are the magnetic resistances of the leakages, and $R_m$ is a core magnetic resistance.  

Superposition principle can be applied and a circuit of Fig. 4.7.2 can be split into two
subcircuits, as presented in Fig. 4.7.2a.  

![Fig-4.7.2a](Figures/Fig-4.7.2a.png)

\begin{equation}\tag{2a.1}
    i_1z_1 = \sideset{_{1\,}}{_m}U = R_m\phi_{21} = R_{l1}\phi_{l1}
\end{equation}

\begin{equation}\tag{2a.2}
    i_2z_2 = \sideset{_{2\,}}{_m}U = R_m\phi_{12} = R_{l2}\phi_{l2}
\end{equation}

Via the above two equations and superposition principle,  

$$ U_m = \sideset{_{1\,}}{_m}U + \sideset{_{2\,}}{_m}U$$

The total flux threading coil 1 is the sum of two components:  

\begin{equation}\tag{2a}
	\phi_{t1} = z_1\phi_1 = z_1\phi_{11}\pm z_1\phi_{12} = z_1\left(\frac{i_1z_1}{R_{l1}} + \frac{i_1z_1}{R_m}\right) \pm z_1\frac{i_2z_2}{R_m} = L_1i_1 \pm Mi_2
\end{equation}

The total flux threading coil 2 is the sum of two components:  

\begin{equation}\tag{2b}
	\phi_{t2} = z_2\phi_2 = z_2\phi_{22}\pm z_2\phi_{21} = z_2\left(\frac{i_2z_2}{R_{l2}} + \frac{i_2z_2}{R_m}\right) \pm z_2\frac{i_1z_1}{R_m} = L_2i_2 \pm Mi_1
\end{equation}

The proof for equation (2a) is:  

$$ \phi_{t1} = z_1\phi_1 \quad\quad\text{where }\; \phi_1 = \phi_{11} \pm \phi_{12}$$

Via equations (1a) and (2a.1),  

$$ \phi_{11} = \phi_{21} + \phi_{l1} = \frac{i_1z_1}{R_m} + \frac{i_1z_1}{R_{l1}}$$

And, via equation (2a.2),  

$$ \phi_{12} = \frac{i_2z_2}{R_m}$$

Via the above three equations,  

$$ \phi_{t1} = z_1\left(\frac{i_1z_1}{R_{l1}} + \frac{i_1z_1}{R_m}\right) \pm z_1\frac{i_2z_2}{R_m}$$

Based on equation (2a),  

\begin{equation}\tag{2a.3}
    \text{a: }\;L_1 = z_1^2\frac{R_m + R_{l1}}{R_{l1}R_m} \quad\quad\text{and}\quad\quad \text{b: }\;M = \frac{z_1z_2}{R_m}
\end{equation}

And, based on equation (2b),  

\begin{equation}\tag{2a.3c}
    L_2 = z_2^2\frac{R_m + R_{l2}}{R_{l2}R_m}
\end{equation}

Constant of proportionality between one coil current and a flux that is produces in the coupled
coil is called the **mutual inductance** $M$. This constant can be expressed by self inductances of individual coils.  

\begin{equation}\tag{3}
	M = k\sqrt{L_1L_2}
\end{equation}

where, $0\le k \le 1$ is the **coefficient of coupling**, a measure of the degree to which the flux
produced by one coil threads another. If there is no coupling then, $k=0\;\implies\;M=0$ . For
tightly coupled coils, which is the most desirable situation, $k=1\;\implies\;M=\sqrt{L_1L_2}$. Via equation 
(2), two-port equations can be obtained.  

\begin{equation}\tag{4a}
    u_1 = \frac{\mathrm{d}\phi_{t1}}{\mathrm{d}t} = L_1i_1'(t)\pm Mi_2'(t)
\end{equation}

\begin{equation}\tag{4b}
    u_2 = \frac{\mathrm{d}\phi_{t2}}{\mathrm{d}t} = L_2i_2'(t)\pm Mi_1'(t)
\end{equation}

The plus or minus is used because the fluxes produced by the coupled coils may point in the same or opposite direction.  In the case of the magnetic circuit shown in figure 4.7.1, the fluxes point in the same direction and thus the plus sign is used in the appropriate equations.

Via equations (2a.3a,c),  

$$ L_1L_2 = z_1^2z_2^2\frac{R_m^2 + R_m(R_{l1} + R_{l2}) + R_{l1}R_{l2}}{R_{l1}R_{l2}R_m^2}$$

And, via the above equation and equations (2a.3b) and (3),  

\begin{equation}\tag{2a.4}
    k = \sqrt{\frac{R_{l1}R_{l2}}{(R_m+R_{l1})(R_m+R_{l2})}}
\end{equation}

### Dot convention
Currents entering the dotted ends are creating additive fluxes. Dotted ends have a positive
voltage at the same time.  

A circuit symbol of coupled coils is presented in Fig. 4.7.3.  

![Fig-4.7.3](Figures/Fig-4.7.3.png)

Using the phasor notation, coupled coils equations are:  

\begin{equation}\tag{5a}
    U_1(j\omega) = j\omega L_1I_1(j\omega) \pm j\omega MI_2(j\omega)
\end{equation}

\begin{equation}\tag{5b}
    U_2(j\omega) = j\omega L_2I_2(j\omega)\pm j\omega MI_1(j\omega)
\end{equation}

#### Example 4.7.1
Consider two practical coupled coils connected in series, as shown in Fig. 4.7.4. Find the
circuit equivalent.  

![Fig-4.7.4](Figures/Fig-4.7.4.svg)

$$ U_1(j\omega) = R_1I_1(j\omega) + j\omega L_1I_1(j\omega) + j\omega MI_2(j\omega)$$

$$ U_2(j\omega) = R_I2(j\omega) + j\omega L_2I_2(j\omega) + j\omega MI_1(j\omega)$$

Via the fact that $I(j\omega) = I_1(j\omega) = I_2(j\omega)$, $U(j\omega) = U_1(j\omega)+U_2(j\omega)$, and the above two equations,  

\begin{equation}\tag{6a}
    U(j\omega) = I(j\omega)\big(R_1 + R_2 + j\omega(L_1+L_2 + 2M)\big)
\end{equation}

\begin{equation}\tag{6b}
    U(j\omega) = I(j\omega)(R_s + j\omega L_s)
\end{equation}

Thus,  

\begin{equation}\tag{7}
    \text{a: }\;L_s = L_1+L_2+2M \quad\quad\text{and}\quad\quad \text{b: }\;R_s = R_1+R_2
\end{equation}

The equivalent series inductance of two coupled coils is sum of self inductances plus the
doubled mutual inductance. For negative coupling the doubled mutual inductance should be
subtracted.

#### Example 4.7.2
A coil of $z=100$ turns connected to a $U=100\mathrm{V}$, $\omega=1000r\,\mathrm{rad}/\mathrm{s}$ supply has the following parameters: $L=1\mathrm{H}$, $R=10\Omega$ . Consider a short-circuit of one turn. Study the effect of such failure.  

A coil with one turn shorted can be considered as pair of coupled coils, as presented in Fig.
4.7.5.  

![Fig-4.7.5](Figures/Fig-4.7.5.png)

Via equation (3.1.14),  

$$ R_m = \frac{z^2}{L} = 10\mathrm{k}\Omega $$

$$ L_1 = \frac{99^2}{R_m}\cdot0.99 = 970.299\mathrm{mH} \quad\quad\text{and}\quad\quad R_1 = 9.9\Omega$$

In [15]:
z = 100; L = 1; R = 10; U = 100; omega = 1000
R_m = z**2/L
L_1 = 0.99*99**2/R_m; R_1=9.9
L_1

0.970299

$$ L_2 = \frac{1}{R_m}\cdot0.01 = 0.1\mathrm{mH} \quad\quad\text{and}\quad\quad R_2 = 0.1\Omega$$

In [16]:
L_2 = 1/R_m; R_2 = 0.1
L_2

0.0001

**Prior** to the short circuit:  

$$ U(j\omega) = RI(j\omega) + j\omega LI(j\omega) \quad\implies\quad I(j\omega) = \frac{U}{R+j\omega L} $$

$$ I(\omega) = \frac{U}{\sqrt{R^2+(\omega L)^2}} \approxeq 99.995\mathrm{mA}$$

In [6]:
from math import sqrt

In [7]:
I = U/sqrt(R**2 + (omega*L)**2)
I

0.09999500037496875

$$ \alpha_i = -\arccos \frac{R}{\sqrt{R^2+(\omega L)^2}} \approxeq -89.43^{\circ}$$

In [8]:
import numpy as np
import matplotlib.pyplot as plt

In [10]:
alpha_i = np.arccos(R / sqrt(R**2+(omega*L)**2))
np.degrees(alpha_i)

89.42706130231652

The dissipated power is:  

\begin{equation}
    P = I^2(\omega)R = \frac{U^2R}{R^2 + (\omega L)^2} = \frac{10^5}{100 + 10^6}\,\mathrm{W} \approxeq 0.1\mathrm{W}
\end{equation}

In [24]:
P = 10**5/(100+10**6)
P

0.0999900009999

**After** the short circuit:  

\begin{equation}\tag{2.1}
    U(j\omega) = R_1I_1(j\omega) + j\omega L_1I_1(j\omega) + j\omega MI_2(j\omega)
\end{equation}

$$ 0 = R_2I_2(j\omega) + j\omega L_2I_2(j\omega) + j\omega MI_1(j\omega)$$

\begin{equation}\tag{2.2}
    I_2(j\omega) = -\frac{j\omega MI_1(j\omega)}{R_2 + j\omega L_2}
\end{equation}

Via equations (2.1,2),  

$$ U(j\omega) = I_1(j\omega)\left(R_1 + j\omega L_1 + \frac{(\omega M)^2}{R_2 + j\omega L_2}\right) = Z(j\omega)I(j\omega)$$

The impedance is given by:  

$$ Z(j\omega) = R_1 + \frac{(\omega M)^2}{R_2^2 + (\omega L_2)^2} + j\omega L_1 - j\frac{\omega L_2}{R_2^2 + (\omega L_2)^2}$$

\begin{equation}\tag{2.3}
    = R_1 + \frac{(\omega M)^2R_2}{R_2^2 + (\omega L_2)^2} + j\omega\left(L_1 - \frac{(\omega M)^2L_2}{R_2^2 + (\omega L_2)^2}\right)
\end{equation}

In [21]:
Z = complex(R_1 + (omega*M)**2*R_2/(R_2**2 + (omega*L_2)**2), \
            omega*L_1 - omega*(omega*M)**2*L_2/(R_2**2 +(omega*L_2)**2))
Z

(495.04949999999997+485.1495j)

In [22]:
sqrt(Z.real**2 + Z.imag**2)

693.1407106789356

\begin{equation}\tag{2.3a}
    Z(\omega) \approxeq 693.14\Omega
\end{equation}

Finally,  

\begin{equation}\tag{2.4}
    I(\omega) = \frac{U}{Z(\omega)} \approxeq 144.27\mathrm{mA}
\end{equation}

In [25]:
I_rms = U/sqrt(Z.real**2 + Z.imag**2)
I_rms

0.14427085072243034

Assuming a coupling coefficient of $k=1$, i.e. assuming no flux leakage,  

\begin{equation}\tag{2.4}
    M = \sqrt{L_1L_2} \approxeq 9.85 \mathrm{mH}
\end{equation}

In [14]:
M = sqrt(L_1*L_2)
M

0.009850375627355537

Finally, the dissipated power is given by:  

\begin{equation}\tag{2.5}
    P = I^2(\omega)\mathbf{Z}_\mathfrak{Re} \approxeq 10.3\mathrm{W}
\end{equation}

In [26]:
I_rms**2*Z.real

10.303999089125245

The impedance magnitude practically has not changed, and consequently rms current remains
unchanged, however the resistance and then dissipated power has increased 100 times !!! This
causes rapid increase of temperature, an isolation melts and next turns are shorted, what
completely destroys (burns) the coil.